nginx-controller/nginx/configurator.go

```diff
@@ -112,7 +176,7 @@ func (cnf *Configurator) generateNginxCfg(ingEx *IngressEx, pems map[string]stri
 	var servers []Server
 
 	for _, rule := range ingEx.Ingress.Spec.Rules {
-		if rule.IngressRuleValue.HTTP == nil {
+		if rule.IngressRuleValue.HTTP == nil && ingEx.Ingress.Annotations["nginx.org/mergible-ingress-type"] != "master" {
```

Comment:
the check for ingEx.Ingress.Annotations["nginx.org/mergible-ingress-type"] != "master" is not required anymore, since you initialize this field `rule.IngressRuleValue.HTTP`  if it is empty.

Reply:
again, can't see why this check is required

Reply:
Removed it. It was before I had initialized `rule.IngressRuleValue.HTTP` for the master.